public/app/core/components/PageActionBar/PageActionBar.tsx

```diff
@@ -1,18 +1,33 @@
 import React, { PureComponent } from 'react';
 
+import { SelectableValue } from '@grafana/data';
 import { LinkButton, FilterInput } from '@grafana/ui';
 
+import { SortPicker } from '../Select/SortPicker';
+
 export interface Props {
   searchQuery: string;
   setSearchQuery: (value: string) => void;
   linkButton?: { href: string; title: string; disabled?: boolean };
   target?: string;
   placeholder?: string;
+  sortPicker?: {
+    onChange: (sortValue: SelectableValue) => void;
+    value?: string;
+    getSortOptions?: () => Promise<SelectableValue[]>;
```

Comment:
Why exactly does this need to be an async getter instead of just a value (`options`)? (If this would be for something in the future, could the call site cater for fetching the options and updating the prop once they are available?)

Reply:
Basically, I just copied the type from SortPicker's props to give freedom to devs on how they want to use SortPicker inside PageActionBar.

SortPicker needs this to be a Promise, because there is a usage where it actually makes a request to the `/sorting` endpoint. 

> (If this would be for something in the future, could the call site cater for fetching the options and updating the prop once they are available?)

I'm not sure if we want to change this behaviour in this PR.

What I think we could do is to define a `sortOptions`  prop in PageActionBar, and pass it to `SortPicker` as 
```
getSortOptions={() => Promise.resolve(sortOptions)}
```, but it makes PageActionBar more restrictive than it could be.

Reply:
I see, thanks for the explanation 👍